core/src/test/java/org/apache/carbondata/core/util/ValueCompressionUtilTest.java

```diff
@@ -0,0 +1,546 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.carbondata.core.util;
+
+import org.apache.carbondata.core.datastorage.store.compression.ValueCompressonHolder;
+import org.apache.carbondata.core.datastorage.store.compression.type.*;
+import org.junit.Test;
+
+import java.nio.ByteBuffer;
+
+import static junit.framework.TestCase.*;
+import static org.apache.carbondata.core.util.ValueCompressionUtil.DataType;
+
+public class ValueCompressionUtilTest {
+
+    @Test
+    public void testGetSize() {
+        DataType[] dataTypes = {DataType.DATA_BIGINT,DataType.DATA_INT,DataType.DATA_BYTE,DataType.DATA_SHORT,DataType.DATA_FLOAT};
+        int[] expectedSizes = {8,4,1,2,4};
+        for(int i =0; i < dataTypes.length; i++) {
+            assertEquals(expectedSizes[i],ValueCompressionUtil.getSize(dataTypes[i]));
+        }
+    }
+
+    @Test
+    public void testToGetCompressedValuesWithCompressionTypeMin_MaxForDataInt() {
+        double[] values = {20.121,21.223,22.345};
+        int[] result = (int[]) ValueCompressionUtil.getCompressedValues(ValueCompressionUtil.COMPRESSION_TYPE.MAX_MIN,values,DataType.DATA_INT,22.3,3);
```

Comment:
The values which are passed are wrong, you are passing decimal values and type passed as `MAX_MIN`, it does not return right result. Please pass the proper values depends up on the compression type.
Please change for other datatypes as well